.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,34 @@
+// For format details, see https://containers.dev/implementors/json_reference/.
+// For config options, see the README at: https://github.com/devcontainers/templates/tree/main/src/ruby
+{
+  "name": "open_street_map",
+  "dockerComposeFile": "compose.yaml",
+  "service": "rails-app",
+  "workspaceFolder": "/workspaces/${localWorkspaceFolderBasename}",
+
+  // Features to add to the dev container. More info: https://containers.dev/features.
+  "features": {
+    "ghcr.io/devcontainers/features/github-cli:1": {},
+    "ghcr.io/devcontainers/features/node:1": {},
+    "ghcr.io/rails/devcontainer/features/activestorage": {},
+    "ghcr.io/rails/devcontainer/features/postgres-client": {}
+  },
+
+  "containerEnv": {
+    "CAPYBARA_SERVER_PORT": "45678",
+    "DB_HOST": "postgres"
+  },
+
+  // Use 'forwardPorts' to make a list of ports inside the container available locally.
+  "forwardPorts": [3000, 5432, 6379],
```

Comment:
3000 = rails app, 5432 = postgres, those seem reasonable. But as I understand it, 6379 is redis, and is that used at all?

Also just to check - do the ports get forwarded to the correct containers - specifically does 5432 get forwarded to the postgres container? I'm no expert on container routing.

Reply:
Good catch. The Redis port is a leftover that I failed to remove. The initial command `bin/rails devcontainer` generates a configuration that includes Redis, which I removed except for that detail.

As for the use of the list of ports... actually I have no idea. I'm no expert either, and I was playing with it now and I can't make any sense of it. I can't access Postgres from outside the container, and if I remove 3000 I can still access the web server with my browser. I'm now wondering if we should remove this line altogether as I don't know what it does at all :thinking: The documentation is not very clear.